Don't copy collection elements in ``test_dataset_collection_hide_originals``

nsoranzo opened the pull request:
Follow-up to https://github.com/galaxyproject/galaxy/pull/16717 .
Minimal performance gain but helps document the parameter use after the change in default value to True.

Also:
- Remove unnecessary check.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

nsoranzo committed:
Don't copy collection elements in ``test_dataset_collection_hide_originals``

Follow-up to https://github.com/galaxyproject/galaxy/pull/16717 .
Minimal performance gain but helps document the parameter use after
the change in default value to True.

mvdbeek commented on lib/galaxy/webapps/galaxy/services/history_contents.py:
```diff
@@ -1270,10 +1270,6 @@ def __create_dataset_collection(
-            if copy_required and not copy_elements:
-                raise exceptions.RequestParameterMissingException(
-                    "copy_elements passed as 'false' but it is required to change specified attributes"
-                )
```
I think the right fix is to set `copy_elements = payload.copy_elements`, you should not change the dbkey without making a copy.

nsoranzo replied:
Isn't `copy_elements = payload.copy_elements or copy_required` enforcing the copy and saving from raising an exception?

mvdbeek replied:
True, but maybe a tad inconsistent / hard to predict for API clients ? It's fine if you want to keep it this way.

nsoranzo replied:
Actually you are right, this would be unexpected behaviour. I've rebased with your suggestion above.

nsoranzo committed:
Remove overriding of ``copy_elements`` parameter

Prevent unexpected behaviour of the API.
Actually raise `RequestParameterInvalidException` exception if this
conflicts with `copy_required` (i.e. dbkey not None).
